src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -538,8 +540,119 @@ export class NotebookModel extends Disposable implements INotebookModel {
 		if (this.inErrorState) {
 			return undefined;
 		}
+
 		let cell = this.createCell(cellType);
+		return this.insertCell(cell, index);
+	}
+
+	public splitCell(cellType: CellType, index?: number, notebookService?: INotebookService): ICellModel | undefined {
+		if (this.inErrorState) {
+			return undefined;
+		}
+
+		let notebookEditor = notebookService.findNotebookEditor(this.notebookUri);
+		let cellEditorProvider = notebookEditor.cellEditors.find(e => e.cellGuid() === this.cells[index].cellGuid);
+		//Only split the cell if the markdown editor is open.
+		//TODO: Need to handle splitting of cell if the selection is on webview
+		if (cellEditorProvider) {
+			let editor = cellEditorProvider.getEditor() as QueryTextEditor;
+			if (editor) {
+				let editorControl = editor.getControl() as CodeEditorWidget;
+
+				let model = editorControl.getModel() as TextModel;
+				let range = model.getFullModelRange();
+				let selection = editorControl.getSelection();
+				let source = this.cells[index].source;
+				let newcell = null, tailcell = null, partial = null;
```

Comment:
We use undefined, not null.

And please give these types so readers know what to expect these to be.

Reply:
Fixed